prometheus_juju_exporter/collector.py

```diff
@@ -150,26 +150,29 @@ def _get_machine_sensible_hostname(self, machine: Dict) -> Union[str, None]:
         field, which seems to exist almost always.
 
         :param dict machine: status information for a machine
-        :return Union[str, None]: a sensible hostname string for the machine if
-            applicable, else None.
+        :return str: a valid identifier string for the machine if
+            applicable, else the string literal "None".
         """
-        sensible_hostname = None
+        self.logger.debug("Trying to find an identifier for machine: %s", str(machine))
+        machine_id = "None"
         for field in ["hostname", "instance-id"]:
             candidate = machine.get(field, None)
             self.logger.debug(
-                "Candidate hostname:[%s] in field:[%s]",
-                candidate,
-                field
+                "Candidate machine id:[%s] in field:[%s]", candidate, field
             )
-            if candidate not in [None, "None"]:
-                sensible_hostname = candidate
+            if candidate not in [None, "None", "pending"]:
+                machine_id = candidate
                 self.logger.debug(
-                    "Selecting sensible hostname:[%s] in field:[%s]",
-                    sensible_hostname,
-                    field
+                    "Selecting machine id:[%s] in field:[%s]",
+                    machine_id,
+                    field,
                 )
                 break
-        return sensible_hostname
+        if machine_id == "None":
```

Comment:
Just a minor note that does not warrant rework: Instead of this `if` statement, you could use `for/else` statement to log errors if the `for` loop completed without encountering `break`.